src/index.ts

```diff
@@ -57,18 +57,36 @@ function getProperty(member: CaseClassMember, isReadonly: boolean, optionalType:
   return gen.property(member.name, type, isOptional, member.desc)
 }
 
-function getNewtype(model: CaseClass): gen.TypeDeclaration {
-  return gen.typeDeclaration(model.name, getType(model.members[0].tpe, false), true)
+interface NewtypeRawDeclaration {
+  name: string
+  declaration: string
+  kind: 'newtype'
+}
```

Comment:
this is the data structure that we manage alongside `gen.TypeDeclaration` and that should be ideally replaced by a custom io-ts-gen Node.